tools/schemacode/bidsschematools/__main__.py

```diff
@@ -32,5 +40,128 @@ def export(ctx, schema, output):
             fobj.write(text)
 
 
+@cli.command("pre-receive-hook")
+@click.option("--schema", "-s", type=click.Path(), help="Path to the BIDS schema")
+@click.option(
+    "--input", "-i", "input_", default="-", type=click.Path(), help="Input file (default: stdin)"
+)
+@click.option(
+    "--output",
+    "-o",
+    "output",
+    default="-",
+    type=click.Path(),
+    help="Output file (default: stdout)",
+)
+def pre_receive_hook(schema, input_, output):
+    """Validate filenames from a list of files against the BIDS schema
+
+    The expected input takes the following form:
+
+    ```
+    bids-hook-v2
+    {"Name": "My dataset", "BIDSVersion": "1.9.0", "DatasetType": "raw"}
+    ignore-pattern1
+    ...
+    ignore-patternN
+    0001
+    .datalad/config
+    .gitattributes
+    CHANGES
+    README
+    dataset_description.json
+    participants.tsv
+    sub-01/anat/sub-01_T1w.nii.gz
+    ...
+    ```
+
+    The header identifies the protocol version. For protocol ``bids-hook-v2``,
+    the second line MUST be the dataset_description.json file, with any newlines removed.
+    The following lines, up to the line containing "0001", are ignore patterns
+    from the .bidsignore file. The lines following "0001" are the filenames to
+    be validated.
+
+    This is intended to be used in a git pre-receive hook.
+    """
+    logger = logging.getLogger("bidsschematools")
+    schema = load_schema(schema)
+
+    # Slurp inputs for now; we can think about streaming later
+    if input_ == "-":
+        stream = sys.stdin
+    else:
+        stream = open(input_)
+
+    first_line = next(stream)
+    if first_line == "bids-hook-v2\n":
+        # V2 format: header line, description JSON, followed by legacy format
+        description_str = next(stream)
+        fail = False
+        try:
+            description: dict = json.loads(description_str)
+        except json.JSONDecodeError:
+            fail = True
+        if fail or not isinstance(description, dict):
+            logger.critical("Protocol error: invalid JSON in description")
+            logger.critical(
+                "Dataset description must be one JSON object, written to a single line"
+            )
+            logger.critical("Received: %s", description_str)
+            stream.close()
+            sys.exit(2)
+    else:
+        # Legacy: ignore patterns, followed by "0001", followed by filenames
+        stream = chain([first_line], stream)
+        description = {}
+
+    dataset_type = description.get("DatasetType", "raw")
+    logger.info("Dataset type: %s", dataset_type)
+
+    ignore = [line.strip() for line in stream if line != "0001\n"]
```

Comment:
```suggestion
    ignore = []
    for line in stream:
        if line == "0001\n":
            break
        else:
            ignore.append(line.strip())
```
This would consume the entire input as ignore patterns.

Reply:
Slightly adjusted your suggestion to skip the redundant else.